lib/src/screens/debug/scheme_management/scheme_manager_detail_screen.dart

```diff
@@ -0,0 +1,253 @@
+          appBar: IrmaAppBar(title: title, hasBorder: false),
+          body: YiviPinScreen(
+            instruction: instruction,
+            pinBloc: EnterPinStateBloc(maxPinSize),
+            maxPinSize: maxPinSize,
+            onSubmit: (pin) => navigator.pop(pin),
+            listener: (context, state) {
+              if (!hasLongPin && state.pin.length == 5) {
+                navigator.pop(state.toString());
+              }
+            },
+          ),
+        ),
+      ),
+    );
+  }
+
+  Future<void> _verifyPin(BuildContext context, String schemeId) async {
+    final repo = IrmaRepositoryProvider.of(context);
+    final navigator = Navigator.of(context);
+
+    final pin = await _requestPin(
+      context,
+      FlutterI18n.translate(context, 'debug.scheme_management.verify_pin.title'),
+      FlutterI18n.translate(
+        context,
+        'debug.scheme_management.verify_pin.content',
+        translationParams: {
+          'scheme': schemeManager.id,
+        },
+      ),
+    );
+    if (pin == null) return;
+
+    repo.bridgedDispatch(AuthenticateEvent(pin: pin, schemeId: schemeId));
+
+    final event = await repo.getEvents().whereType<AuthenticationEvent>().first;
+    navigator.pop();
+
+    if (event is AuthenticationErrorEvent) {
+      navigator.push(MaterialPageRoute(
+        builder: (context) => ErrorScreen(
+          details: event.error.toString(),
+          onTapClose: () => navigator.pop(),
+        ),
+      ));
+    } else {
+      showSnackbar(
+        context,
+        event is AuthenticationFailedEvent
+            ? FlutterI18n.translate(
+                context,
+                'debug.scheme_management.verify_pin.failed',
+                translationParams: {
+                  'remainingAttempts': event.remainingAttempts.toString(),
+                  'blockedDuration': event.blockedDuration.toString(),
+                },
+              )
+            : FlutterI18n.translate(context, 'debug.scheme_management.verify_pin.success'),
+      );
+    }
+  }
+
+  Future<void> _onActivateScheme(BuildContext context) async {
+    final language = FlutterI18n.currentLocale(context)?.languageCode ?? 'en';
+
+    final pin = await _requestPin(
+      context,
+      FlutterI18n.translate(context, 'debug.scheme_management.request_pin.title'),
+      FlutterI18n.translate(
+        context,
+        'debug.scheme_management.request_pin.content',
+        translationParams: {
+          'scheme': schemeManager.id,
+        },
+      ),
+    );
+    if (pin == null) return;
+
+    final navigator = Navigator.of(context);
+    final email = await navigator.push(
+      MaterialPageRoute(
+        builder: (context) => ProvideEmailScreen(
+          onEmailProvided: (email) => navigator.pop(email),
+          onEmailSkipped: () => navigator.pop(''),
+          onPrevious: () => navigator.pop(),
+        ),
+      ),
+    );
+    if (email == null) return;
+
+    navigator.pop();
```

Comment:
I think this also leads to a double pop, but I did not test this case in particular.